packages/webview_flutter/webview_flutter_platform_interface/lib/src/types/web_resource_response.dart

```diff
@@ -0,0 +1,56 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'web_resource_request.dart';
+
+/// Contains information about the response for a request.
+///
+/// Platform specific implementations can add additional fields by extending
+/// this class.
+///
+/// This example demonstrates how to extend the [WebResourceResponse] to
+/// provide additional platform specific parameters.
+///
+/// When extending [WebResourceResponse] additional parameters should always
+/// accept `null` or have a default value to prevent breaking changes.
+///
+/// ```dart
+/// class AndroidWebResourceResponse extends WebResourceResponse {
+///   WebResourceResponse._({
+///     required WebResourceResponse response,
+///   }) : super(
+///     request: response.request,
+///     headers: response.headers,
+///   );
+///
+///   factory AndroidWebResourceResponse.fromWebResourceResponse(
+///     WebResourceResponse response, {
+///     Uri? historyUrl,
+///   }) {
+///     return AndroidWebResourceResponse._(response, historyUrl: historyUrl);
+///   }
+///
+///   final Uri? historyUrl;
+/// }
+/// ```
+@immutable
+class WebResourceResponse {
+  /// Used by the platform implementation to create a new [WebResourceResponse].
+  const WebResourceResponse({
+    required this.request,
+    required this.statusCode,
+    this.headers = const <String, String>{},
+  });
+
+  /// The request
+  final WebResourceRequest request;
```

Comment:
@stuartmorgan I believe you missed my response here: https://github.com/flutter/packages/pull/4025#discussion_r1327890280

I think this should be a `String? url`. Based on the platform APIs:

Android only provides a url through `WebResourceRequest.url`: https://developer.android.com/reference/android/webkit/WebViewClient#onReceivedHttpError(android.webkit.WebView,%20android.webkit.WebResourceRequest,%20android.webkit.WebResourceResponse)
iOS only provides a url through `NSHTTPURLResponse.URL`: https://developer.apple.com/documentation/webkit/wknavigationdelegate/1455643-webview?language=objc

I believe the latter accounts for redirects while the former doesn't.

Reply:
I see, I was basing my earlier comment on the code comment, which said it was the request URI. If what we really have a response URI that we populate differently on different platforms, then a `String? uri` with clearer documentation (e.g., "The URI that this response is associated with." or something) SGTM.